src/portal.py

```diff
@@ -198,14 +207,47 @@ def portal_update(
             (firstname + " " + lastname).strip()
         ]
 
+    new_mails = current_user["mail"]
+
+    if mainemail is not None:
+        is_allowed_to_edit_main_email = portal_settings["allow_edit_email"]
+        if is_allowed_to_edit_main_email != 1:
+            raise YunohostValidationError("mail_edit_operation_unauthorized")
+
+        if mainemail not in new_mails:
+            local_part, domain = mainemail.split("@")
+            if local_part in ADMIN_ALIASES:
+                raise YunohostValidationError("mail_unavailable")
+
+            try:
+                _get_ldap_interface().validate_uniqueness({"mail": mainemail})
+            except YunohostError:
+                raise YunohostValidationError(
+                    "mail_already_exists", mail=mainemail
+                )
+
+            if domain not in domains or not user_is_allowed_on_domain(username, domain):
+                raise YunohostValidationError("mail_alias_unauthorized", domain=domain)
+            new_mails[0] = mainemail
+        else:
+            # email already exist in the list we just move it on the first place
+            new_mails.remove(mainemail)
+            new_mails.insert(0, mainemail)
```

Comment:
So if we have in the form
```
Mail: sam@example.com
Mail alias: 
```
And we replace `sam@example.com` by `camille@example.com`, we get
```
Mail: camille@example.com
Mail alias: sam@example.com
```
instead of
```
Mail: camille@example.com
Mail alias: 
```

Reply:
```suggestion
            new_mails = [mainemail] + new_mails[1:]
```